.github/scripts/trymerge.py

```diff
@@ -868,10 +869,25 @@ def merge_on_green(pr_num: int, repo: GitRepo, dry_run: bool = False, timeout_mi
         current_time = time.time()
         elapsed_time = current_time - start_time
 
-
         pr = GitHubPR(org, project, pr_num)
         try:
-            return pr.merge_into(repo, dry_run=dry_run)
+            if all_green:
+                checks = pr.get_checkrun_conclusions()
+                pending_checks = []
+                failed_checks = []
+                for check in checks:
+                    if checks[check] is None:
+                        pending_checks.append(check)
+                    elif checks[check] != 'SUCCESS':
+                        failed_checks.append(check)
+                if len(failed_checks) > 0:
+                    raise RuntimeError(f"Failed to merge due to failing checks, {','.join(failed_checks)}")
+                if len(pending_checks) > 0:
+                    reject_reason = f"Refusing to merge as mandatory check(s) {','.join(pending_checks)} are missing"
+                    raise MandatoryChecksMissingError(reject_reason)
+                pr.merge_into(repo, dry_run=dry_run)
```

Comment:
Oh I see, this calls the code again. In that case go for it lol, but Eli's comment about the API limit is important since we did have SEVs about our GITHUB_TOKEN getting overused.